lib/snmp.js

```diff
@@ -715,6 +716,48 @@ Session.prototype.getNext = function (options, callback) {
     self.sendMsg(pkt, options, callback);
 };
 
+// Shortcut to create a GetBulkRequest and send it, while registering a callback.
+// Needs `options.oid` to be an OID in array form.
+
+Session.prototype.getBulk = function (options, callback) {
+    var self = this, pkt;
+
+    defaults(options, self.options);
+    parseOids(options);
+
+    if (!options.oid) {
+        return callback(null, []);
+    }
+
+    pkt = new Packet();
+    pkt.community = options.community;
+    pkt.version = options.version;
+    // For GetBulkRequest PDUs the error and errorIndex are overridden
+    // and represent the non-repeaters and max-repetitions variables
+    // non-repeaters is the number of varbinds which are not repeated
+    // max-repetitions is the number of next variables that should be fetched
+    pkt.pdu.error = 0;
+    pkt.pdu.errorIndex = options.max_repetitions;
+    pkt.pdu.type = 5;
+    pkt.pdu.varbinds[0].oid = options.oid;
+    self.sendMsg(pkt, options, callback);
+};
+
+// Helper to check whether `oid` in inside the tree rooted at
+// `root` or not.
+var inTree = function (root, oid) {
```

Comment:
because inTree() now is used by both the pre-existing getSubtree() and the new getSubtreeBulk().